### ISSUE TITLE
Syphon Unity 5.5 / OpenGLCore problems

### ISSUE THREAD

**Issue by benkuper** [Syphon/Unity3D#26]
Hello, this problem has already been discussed in a closed thread, but as Unity 5.5 doesn't support OpenGL2 anymore, the workaround consisting in setting OpenGL2 as the renderer in the player prefs doesn't work anymore.

Is the support for OpenGLCore or Metal planned ?

Thank you

**Comment by vade** [comment]
We have a branch with experimental Core Profile support (which I am behind on, thanks for reminding me) - and Metal is a possibility although not a priority at this point.

**Comment by benkuper** [comment]
Thank you, did not see that one.
Unfortunately, it seems that it still doesn't work with Unity 5.5, maybe some bad config on my project, if you manage to get U5.5 working with syphon, let us know !

**Comment by eccoscreen** [comment]
where can i find this branch at? looking to get syphon working in 5.5 but it's not showing up

**Comment by axistudio** [comment]
@eccoscreen @benkuper @vade @codeanticode @astellato 
check this one [](https://github.com/keijiro/Funnel/tree/gllegacy)
it works with 5.5

**Comment by benkuper** [comment]
Yeah Funnel is the way to go if you want a working server from Unity.
In my case, i'm interested in a Syphon client in unity (getting an openFramework shared texture into Unity), and Funnel only handles sending texture, not receiving them.

**Comment by axistudio** [comment]
thats really bad, I didn't try receive texture in unity  .i was wondering any core coder can help us to update syphon into unity 5. We can crowdfunding little bit, get someone to help. I personally are willing to donate 100 already.

**Comment by vade** [comment]
Hi all - apologies. Theres some testing left to do with the Core Profile branch of Syphon. Ive been distracted with paying bills, but its on our to-do. Im off for a weekend residency where I can actually spend some time finalizing some things for Core Profile Syphon. I pledge to give it a little bit of time. That doesnt mean Unity plugin will just start working, because Syphon needs to support some more modern OpenGL, and then the Unity Plugin will need to be re-written to take advantage of that.

I believe @brianchasalow is interested in Unity 5.5 + Support, and was one of the core authors behind the original legacy GL implementation. Hopefully he, or one of the other developers from other Syphon / Unity implementations can hop on to finish up the latter end of the Unity plugin integration once we (I) have Syphon Core Profile support polished a bit more.

No promises, life is crazy right now! (And honestly, money helps me feel better about allocating time to do this, but its something both @bangnoise and I want to have finished). The ball is clearly in my court right now however.

**Comment by Davidcc** [comment]
Hi, we are desperate to get a syphon source into Unity 5.5. Any hope that this will work?

**Comment by rom-io** [comment]
@axistudio I can add $150 to the pot to speed things up if I get an invoice from @vade / @brianchasalow .
Need the client-end/texture fixed for a project.

**Comment by freekatet** [comment]
Did anything happened?

**Comment by peteforde** [comment]
Things are tight but I'd happily contribute an additional $100 to the pot if we can get movement on this.

Big :heart: to @vade and the rest of the team.

**Comment by axistudio** [comment]
shall we do a fund campaign in the group of facebook ?we can donate on the syphon website.

**Comment by peteforde** [comment]
So, for what it's worth I was able to get Funnel to work great with Unity 2017.1, which comes as a huge relief. Of course, Funnel only handles the server side of the transaction; those looking to use a Syphon stream as a texture are still out of luck.

I agree that Funnel should just be merged into the main Syphon library.

**Comment by grigM** [comment]
Hello Syphon TEAM! @bangnoise @vade @brianchasalow 
First of all, i wan't to thank you all for this amazing technology!
And second, i wan't to ask.. 
Is there any chance to see Syphon Unity plugin update for OpenGLCore in future ? 
I can help with any testing or documentation if you need, or make small donation ($100)

**Comment by simonjc123** [comment]
I second the above, Syphon is awesome, use it for live interactive events all the time, but also us Unity for live interactives too - this plug in really is the missing link in my workflow. Syphoner is great, but direct Unity to Syphon would be perfect!
As above - let me know if people as clubbing together for dev costs.

**Comment by brianchasalow** [comment]
convince @vade or @bangnoise to port Syphon to Metal and I can make it work in Unity again. GL Core support is a bit of a nightmare, and Apple probably doesn't want to support GL going forward anyway. I'm not sure what it'd take to port to Metal, but I know for a fact that it theoretically could.

**Comment by Ubik3D** [comment]
I would back a Metal supported version of Syphon with server and client functionality. If @vade or @bangnoise are up for the task? Seems like several users want to back something like this?  @axistudio @grigM @peteforde @Davidcc ?

**Comment by simonjc123** [comment]
I would too! @vade and @bangnoise this would be such a useful patch - I do a lot of live interactive instals and this is basically the missing link in my workflow! S

**Comment by felixbinder** [comment]
https://github.com/keijiro/KlakSyphon This will work with the newest version of Unity for both client and server. It needs a Metal based project though
